dot/core/interface.go

```diff
@@ -47,6 +47,7 @@ type BlockState interface {
 	StoreRuntime(common.Hash, runtime.Instance)
 }
 
+//go:generate mockgen -destination=./mocks/storage_state_mock.go -package mocks . StorageState
 //go:generate mockery --name StorageState --structname StorageState --case underscore --keeptree
```

Comment:
Why do we have both? Also should we have the `StorageState` mock in a `_test.go` file? I'm not sure where it's used really though.

Reply:
In fact, we don't need both, remove and kept just the mockery. Change to mockgen will need a large refactor that is not part of this PR